src/request-queue.ts

```diff
@@ -1,21 +1,31 @@
-import { QueuedRequestOptions } from './types'
+import { QueuedRequestWithOptions, RequestQueueConfig } from './types'
 import { each } from './utils'
 
 import { isArray, isUndefined } from './utils/type-utils'
+import { clampToRange } from './utils/number-utils'
+
+export const DEFAULT_FLUSH_INTERVAL_MS = 3000
```

Comment:
Feels like a solution that doesn't actually relate to the problem...

The problem sounds more to me like customers want a way to flush data and wait for the response before unloading?

Reply:
In the specific case they have an auto-refresh after 2 seconds, so they wouldn't want to delay too long

it's also a much bigger change to wait on refreshes...

what I'd love to do instead is have this queue store to local storage as well as memory, and then this all becomes much safer

(so, yep, this is for sure a workaround I can give this customer quickly, much more than something I'd expect many people to set)